minitwit/src/main/java/WebApplication.java

```diff
@@ -556,4 +570,22 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
         });
         return filteredMessages.toList();
     };
+
+    public static Route serveSimRegister = (Request request, Response response) -> {
+        var json = gson.fromJson(request.body(), HashMap.class);
+        var username = String.valueOf(json.get("username"));
+        var email = String.valueOf(json.get("email"));
+        var password = String.valueOf(json.get("pwd"));
+
+        var error = register(username, email, password, password);
```

Comment:
Does the simulator only provide one password, as opposed to the UI where you need to repeat your password, thus providing two? Bit strange, but if they made it like this, then I suppose what you did here is correct. 😛

Reply:
Yesp!